core/src/test/java/apoc/util/UtilsTest.java

```diff
@@ -291,6 +302,129 @@ public void testMd5() {
+            final var node = tx.createNode(Label.label("HashFunctionsAreStable"));
+            final var rel = node.createRelationshipTo(node, RelationshipType.withName("R"));
+            for (int i = randStorables.size() - 1; i >= 0; --i) {
+                node.setProperty("p" + i, randStorables.get(i));
+                rel.setProperty("p" + i, randStorables.get(i));
+            }
+            tx.commit();
+        }
+
+        try (final var tx = db.beginTx()) {
+            for (int i = 0; i < randStorables.size(); ++i) {
+                final var value = randStorables.get(i);
+                final var query =
+                        """
+                        match (n:HashFunctionsAreStable)
+                        return
+                          apoc.util.%s([n.%s]) as hash1,
+                          apoc.util.%s([$value]) as hash2,
+                          apoc.util.%s($list) as hash3
+                        """
+                                .formatted(hashFunc, "p" + i, hashFunc, hashFunc);
+                final var params = Map.of("value", value, "list", List.of(value));
+                assertStableHash(seed, value, tx, query, params);
+            }
+
+            final var mapQuery =
+                    """
+                    match (n:HashFunctionsAreStable)
+                    return
+                      apoc.util.%s([properties(n)]) as hash1,
+                      apoc.util.%s($value) as hash2
+                    """
+                            .formatted(hashFunc, hashFunc);
+            assertStableHash(seed, randMap, tx, mapQuery, Map.of("value", List.of(randMap)));
+
+            final var listCypher = IntStream.range(0, randStorables.size())
+                    .mapToObj(i -> "n.p" + i)
+                    .collect(Collectors.joining(","));
+            final var listQuery =
+                    """
+                match (n:HashFunctionsAreStable)
+                return
+                  apoc.util.%s([%s]) as hash1,
+                  apoc.util.%s($value) as hash2
+                """
+                            .formatted(hashFunc, listCypher, hashFunc);
+            assertStableHash(seed, randStorables, tx, listQuery, Map.of("value", randStorables));
+
+            final var listOfListsQuery =
+                    """
+                match (n:HashFunctionsAreStable)
+                return
+                  apoc.util.%s([[%s]]) as hash1,
+                  apoc.util.%s([$value]) as hash2
+                """
+                            .formatted(hashFunc, listCypher, hashFunc);
+            assertStableHash(seed, randStorables, tx, listOfListsQuery, Map.of("value", randStorables));
+
+            final var nodesHashQuery =
+                """
+                match (a:HashFunctionsAreStable)-[r]->(b:HashFunctionsAreStable)
+                return
+                  apoc.util.%s([a, b, r, [a, b, r]]) as hash1,
+                  apoc.util.%s([a, b, r, [a, b, r]]) as hash2
```

Comment:
Is it intentional the hash1 and hash2 is the same in this query? Otherwise, you could remove the hash2 line

Reply:
It can be removed, it's pretty useless. :)

Reply:
Removed